pandas/core/indexes/interval.py

```diff
@@ -1120,9 +1120,19 @@ def interval_range(
             # error: Argument 1 to "maybe_downcast_numeric" has incompatible type
             # "Union[ndarray[Any, Any], TimedeltaIndex, DatetimeIndex]";
             # expected "ndarray[Any, Any]"  [
+            if isinstance(start, int) or isinstance(end, int):
+                dtype = np.dtype("int64")
+            else:
+                dtype = start.dtype if start.dtype == end.dtype else np.dtype("int64")
             breaks = maybe_downcast_numeric(
                 breaks,  # type: ignore[arg-type]
-                np.dtype("int64"),
+                dtype,
+            )
+            return IntervalIndex.from_breaks(
+                breaks,
+                name=name,
+                closed=closed,
+                dtype=IntervalDtype(subtype=breaks.dtype, closed=closed),
```

Comment:
Can you just pass a `dtype` to the `IntervalIndex.from_breaks` below?

Reply:
Yes, I can do that!

Reply:
All changes have been implemented and tests are passed. Unit test is added as well!